bimsb/packages/bioinformatics-nonfree.scm

```diff
@@ -1882,3 +1883,72 @@ MACE is a bioinformatics tool dedicated to analyze ChIP-exo data.")
     ;; others say "free for all use", others say "GPL version 1.3" (?), yet
     ;; others say "Artistic license".
     (license license:gpl2+)))
+
+;; Although this program is released under the GPL it depends on
+;; ViennaRNA and bowtie1, which are non-free software.
+(define-public mirdeep2
+  (package
+    (name "mirdeep2")
+    (version "0.0.92")
+    (source (origin
+              (method git-fetch)
+              (uri (git-reference
+                   (url "https://github.com/rajewsky-lab/mirdeep2.git")
+                   (commit (string-append "v" version))))
+              (file-name (string-append name "-" version "-checkout"))
+              (sha256
+               (base32
+                "1gasx6vb38n4vbjbn5s7ch2rpflgf2crdd8mdp2xm571w2j01wmm"))))
+    (build-system gnu-build-system)
+    (arguments
+     `(#:phases
+       (modify-phases %standard-phases
+         (delete 'configure)
+         (delete 'build)
+         (delete 'check)
+         (replace 'install
+           (lambda* (#:key inputs outputs #:allow-other-keys)
+             (let ((target (string-append (assoc-ref outputs "out")
+                                          "/bin")))
+               (with-directory-excursion "src"
+                 (copy-recursively "." target))
+               (copy-file "Rfam_for_miRDeep.fa"
+                          (string-append  target "/../Rfam_for_miRDeep.fa"))
+               (with-output-to-file (string-append target "/../install_successful")
+                 (const #t)))  ;; simply touch the file
+             #t))
+         (add-after 'install 'wrap-perl-scripts
+          (lambda* (#:key inputs outputs #:allow-other-keys)
+            (let ((out (assoc-ref outputs "out")))
+              ;; Make sure perl scripts find all perl inputs at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PERL5LIB" ":" prefix
+                              (,(getenv "PERL5LIB")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "miRDeep2.pl"))
+              ;; Make sure perl scripts find all input binaries at runtime.
+              (for-each (lambda (prog)
+                          (wrap-program (string-append out "/bin/" prog)
+                            `("PATH" ":" prefix
+                               (,(getenv "PATH")))))
+                        '("make_html2.pl"
+                          "make_html.pl"
+                          "mapper.pl"
+                          "miRDeep2_core_algorithm.pl"
+                          "miRDeep2.pl"
+                          "prepare_signature.pl"
+                          "quantifier.pl"))
+              #t))))))
+    (inputs
+      `(("bowtie1" ,bowtie1)
+       ("perl-pdf-api2" ,perl-pdf-api2)
+       ("perl" ,perl)
```

Comment:
Not sure if I need this having a perl-library as input.

Reply:
You would need "perl" among the inputs if any of the scripts have a Perl shebang (even if they use the /usr/bin/env trick). Guix would then automatically patch the shebang for you so that the particular variant of Perl from the inputs is used.

Since this package provides Perl scripts you will need Perl as an input.  The Perl module alone does not propagate Perl -- it just uses it to be built.